content/6_Advanced/Parallel_Binary_Search.mdx

```diff
@@ -0,0 +1,132 @@
+---
+id: parallel-binary-search
+title: 'Parallel Binary Search'
+author: Muaath Alqarni
+description: 'Answering multiple binary search queries parallelly'
+prerequisites:
+  - binary-search
+frequency: 0
+---
+
+
+## Resources
+
+<Resources>
+	<Resource source="CF" title="Parallel Binary Seach [tutorial]" url="45578" />
+</Resources>
+
+<FocusProblem problem="sam" />
+
+## Solution - New Roads Queries
+Let's think how to binary search a single query.
+It's clear that we can use DSU with linear search, but also we can do binary search with DSU in $O(N \cdot \log_2{N})$.
+
+Now let's use parallel binary search:
+Sweep through the array of edges and union them in the DSU,
+whenevery you find a query its median = $i$, if the two nodes where connected, it means shrink to left, otherwise shrink to right.
+
+You need to do the sweep $O(\log_2{N})$ times, so you know the exact time the nodes where connected.
```

Comment:
use "we" instead of "you". No real reason for this change, but it's to keep modules consistent

Also what does "so you know the exact time the nodes where connected." means? What does "time" represent?

Reply:
I should replace time with "amount of days", same as in the problem.